File: backend/src/main/java/com/tientoan21/entity/Voucher.java
package com.tientoan21.entity;

import com.tientoan21.enums.DiscountType;
import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Table(name = "vouchers")
public class Voucher extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String code;
    @Enumerated(EnumType.STRING)
    private DiscountType discountType;
    private BigDecimal discountValue;
    private BigDecimal minOrderValue;
    private BigDecimal maxDiscount;
    private LocalDateTime expiryDate;
    private Integer usageLimit;
    @Builder.Default
    private Integer usedCount = 0;

}

File: frontend/src/types/voucher.ts
export interface VoucherRequest {
    code: string;
    discountType: 'PERCENT' | 'FIXED';
    discountValue: string;
    minOrderValue: string;
    maxDiscount: string;
    expiryDate: Date;
    usageLimit: number;
}
export interface VoucherResponse extends VoucherRequest {
    id: number;
    calculatedDiscount: string;
}